scripts/pre_commit_linter.py

```diff
@@ -3391,4 +3390,5 @@ def main(args=None):
 
 
 if __name__ == '__main__':
+    install_third_party_libs.main(args=[])
```

Comment:
Since python files are executed when they are imported, importing pre_commit_linter causes the "install_third_party_libs" script to be executed, which is unintended behavior -- we only want the script to run when the file is executed directly, which is why this was moved to this location. This unintended behavior leads to a race condition, since Circle CI runs tests on a Linux machine and parallelizes the tests, if multiple tests are importing pre_commit_linter, then multiple tests are instaling third party libs at the same time. "install_third_party_libs" calls setup_gae.py, which iterates through the files in the filesystem and deletes any extra .pyc files. Since the file system is shared by each thread/process running each individual test, a race condition is introduced with respect to getting all the files and deleting them (behavior was indeed exhibited on CircleCi, which is what prompted this change).

Reply:
This is fine with me.

Reply:
Sorry, I misinterpreted your statement above. The parallelization that occurs is at the test file level, not at the test level, so the file is only imported once.